### ISSUE TITLE
`plus/indent-binary-ops`: False positive in specific case

### ISSUE THREAD

**Issue by Ky6uk** [eslint-stylistic/eslint-stylistic#406]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Code like this should be correct:

```ts
const woof = computed(() => keys.value.filter(
  ({ type }) => type === 'bark' ||
    type === 'pooque' ||
    type === 'srenque' ||
    type === 'meow'
));
```

But the rule returns:

```
3:1  error  Expected indentation of 2 spaces  @stylistic/plus/indent-binary-ops
4:1  error  Expected indentation of 6 spaces  @stylistic/plus/indent-binary-ops
5:1  error  Expected indentation of 6 spaces  @stylistic/plus/indent-binary-ops
```

### Reproduction

https://stackblitz.com/edit/plusindent-binary-ops-bug?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
